monkey/common/network/network_utils.py

```diff
@@ -9,11 +9,14 @@ def get_my_ip_addresses() -> Sequence[IPv4Address]:
     return [interface.ip for interface in get_network_interfaces()]
 
 
-def get_network_interfaces() -> List[IPv4Interface]:
+def get_network_interfaces(include_link_local_ips: bool = False) -> List[IPv4Interface]:
```

Comment:
I'd say remove this parameter. Link-local is non-routable, I don't think we'll need it. And if we do need it for some special case, it's better to add a `get_linklocal_interfaces` method